### PR TITLE
add files for prow jenkins e2e job that allow tagging image to be tes…

### PR DESCRIPTION
**Pull request by gabemontero**
…ted into test cluster

for ci-operator/prow extended test invocation

see https://github.com/openshift/release/pull/947

### CODE REVIEW

**Comment by gabemontero** [comment]
@openshift/sig-developer-experience fyi

File: test-e2e/tag-in-image.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+oc tag --source=docker $IMAGE_PREPARE openshift/jenkins:2
```
Comment (bparees) [inline test-e2e/tag-in-image.sh]: maybe a comment about how IMAGE_PREPARE is set/inherited from ci-operator

(and in general what this script/image is for)

Comment (gabemontero) [reply test-e2e/tag-in-image.sh]: script comment explaining things pushed

File: test-e2e/tag-in-image.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# to facilitate the use of test suites defined in openshift/origin
+# (https://github.com/openshift/origin) against images produced
+# by builds of this repository.
+#
+# 
+# The flow is as follows:
+#
+# - ci-operator takes a git branch ref for this repo and config specific to this repo
+#   and produces the jenkins related images from this repo using that git branch ref,
+#   via the openshift docker build startegy (eating our dog food)
```
Comment (bparees) [inline test-e2e/tag-in-image.sh]: strategy

File: test-e2e/tag-in-image.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#   commands that can massage the set up of the test cluster prior to
+#   the running of the tests ... to "prepare" the cluster after it is installed
+
+# - Those plug points are specified via the ci-operator config for
+#   jenkins stored in http://github.com/openshift/release
+
+# - In particular, the test image used to run the tests is defined,
+#   and for the case of jenkins, we take a base test image define
+#   by the openshift internal CI infrastructure, and extend it 
+#   via an openshift docker startegy build, using the Dockerfile
```
Comment (bparees) [inline test-e2e/tag-in-image.sh]: strategy

File: test-e2e/tag-in-image.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# - The template has some plug points (via env vars) to run customizable
+#   commands that can massage the set up of the test cluster prior to
+#   the running of the tests ... to "prepare" the cluster after it is installed
+
+# - Those plug points are specified via the ci-operator config for
+#   jenkins stored in http://github.com/openshift/release
+
+# - In particular, the test image used to run the tests is defined,
+#   and for the case of jenkins, we take a base test image define
```
Comment (bparees) [inline test-e2e/tag-in-image.sh]: defined

Comment (gabemontero) [reply test-e2e/tag-in-image.sh]: typo fixes pushed

**Pull request merged**